drupal_seamless_cilogon.module

```diff
@@ -0,0 +1,166 @@
+ * on the access-ci.org domain, include that in the cookie as well.
+ */
+function drupal_seamless_cilogon_user_login($args)
+{
+
+  $seamless_debug = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_debug', true);
+
+  $domain_verified = verify_domain();
+  
+  if ($seamless_debug) {
+    $msg =  __FUNCTION__ . "() - domain_verified = " . print_r($domain_verified, true) 
+      . ' -- ' . basename(__FILE__) . ':' . __LINE__ ;  
+    \Drupal::messenger()->addStatus($msg);
+    \Drupal::logger('seamless_cilogon')->debug($msg);
+  }
+
+  // only start this flow if we the domain is verified
+  if (!$domain_verified) {
+    return;
+  }
+
+  $site_name = \Drupal::config('system.site')->get('name');
+
+  $cookie_name = \Drupal::state()->get(
+    'drupal_seamless_cilogon.seamless_cookie_name',
+    DrupalSeamlessCilogonEventSubscriber::seamlessCookieName
+  );
+
+  $cookie_value = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_value', null);
+  if (strlen($cookie_value) == 0) {
+    $cookie_value =  "INITIAL_DOMAIN=$site_name";
+  }
+
+  $cookie_expiration = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_expiration', null);
+  $cookie_expiration = strlen($cookie_expiration) == 0
+    ? strtotime('+18 hours')  // default to +18 hours
+    : strtotime($cookie_expiration);  // use value from form
+  
+  // Set the cookie domain (unless debugging)
+  $cookie_domain = $seamless_debug ? null : 'access-ci.org';
+
+  // set the cookie. 
+  setcookie(
+    $cookie_name,
+    $cookie_value,
+    $cookie_expiration,
+    '/', // path
+    $cookie_domain, // domain
+  );
+
+  if ($seamless_debug) {
+    $msg =  __FUNCTION__ . "() - set cookie, name = $cookie_name, value = $cookie_value, expiration = $cookie_expiration "
+      . " = " . date("Y-m-d H:i:s", $cookie_expiration) . ", domain = $cookie_domain"
+      . ' -- ' . basename(__FILE__) . ':' . __LINE__ ;  
+    \Drupal::messenger()->addStatus($msg);
+    \Drupal::logger('seamless_cilogon')->debug($msg);
+  }
+}
+
+/**
+ * Implements hook_user_logout().  When logging out from ACCESS Support, remove the cookie and 
+ * redirect to https://cilogon.org/logout/.
+ */
+function drupal_seamless_cilogon_user_logout($account)
+{
+  if (!verify_domain()) {
+    return;
+  }
+  
+  $cookie_name = \Drupal::state()->get(
+    'drupal_seamless_cilogon.seamless_cookie_name',
+    DrupalSeamlessCilogonEventSubscriber::seamlessCookieName
+  );
+
+  // Set cookie in the past and then remove it.
+  setcookie(
+    $cookie_name,
+    '',
+    strtotime('-1 hour'),
+    '/'
+  );
+  unset($_COOKIE[$cookie_name]);
+
+
+  $seamless_debug = \Drupal::state()->get('drupal_seamless_cilogon.seamless_cookie_debug', true);
+  if ($seamless_debug) {
+    $msg =  __FUNCTION__ . "() - unset cookie, name = $cookie_name"
+      . ' -- ' . basename(__FILE__) . ':' . __LINE__ ;  
+    \Drupal::messenger()->addStatus($msg);
+    \Drupal::logger('seamless_cilogon')->debug($msg);
+  }
+}
+
+
```

Comment:
Hi @a-pasquale -- I have questions about `verify_domain()`.  Per design, both the login & logout hooks will only be run if `verify_domain()` returns true.  And I understand that this module may be used by multiple sites.  

Questions:

- we return `false` if the `domain` module is not installed.  This is because operations (or some other sites) might not have this module installed at all, but we are using it to verify the current domain is `access-support` -- correct?

Reply:
per our chat, i updated `verify_domain()` to return _true_ if the domain module is not installed.